### PR TITLE
test: Add tests for shallowMount issue with dynamic components

### PR DESCRIPTION
**Pull request by phobetron**
This PR adds a minimal set of tests to describe issue #1277, which regards how dynamic components, when shallow mounted, are stubbed with the name of a computed property that returns a component's name instead of the component itself.

The test component outlines the most extreme error case: a computed property does not need to be referenced in order to trigger the issue, it merely needs to both exist and return the component that is directly referenced.

This PR now also implements a possible simple fix for the issue.

cc: @cexbrayat

### CODE REVIEW

File: tests/shallowMount.spec.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { defineAsyncComponent, defineComponent } from 'vue'
 import { mount, shallowMount, VueWrapper } from '../src'
 import ComponentWithChildren from './components/ComponentWithChildren.vue'
 import ScriptSetupWithChildren from './components/ScriptSetupWithChildren.vue'
+import ScriptSetupWithBuiltinComponent from './components/ScriptSetupWithBuiltinComponent.vue'
```
Comment (cexbrayat) [inline tests/shallowMount.spec.ts:5]: nit: I would rename to `DynamicComponentWithComputedProperty` or something like that.

File: tests/components/ScriptSetupWithBuiltinComponent.vue [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <component :is="Hello" />
```
Comment (cexbrayat) [inline tests/components/ScriptSetupWithBuiltinComponent.vue:2]: shouldn't that use `computedProperty`?

Comment (phobetron) [reply tests/components/ScriptSetupWithBuiltinComponent.vue:2]: @cexbrayat This is what I meant when I added the note. It does not matter whether you _use_ the computed property, its name is still what `shallowMount` will give to `component`. That's how weird this bug is.

Comment (phobetron) [reply tests/components/ScriptSetupWithBuiltinComponent.vue]: I'll add a third test to better outline the issue.

**Comment by cexbrayat** [comment]
Hmm strange indeed, thanks for the repo 👍

If you feel adventurous, you can try to inspect what's going on when the tests run, and you may be able to fix the issue: we would be very happy to accept the PR!

**Comment by phobetron** [comment]
@cexbrayat The issue was exactly where @freakzlike had suggested. My fix is to ensure `getComponentNameInSetup` doesn't get tripped up by any getter functions that may return a component object. All the tests now pass. Please let me know what you think.

**Review by lmiller1990 — APPROVED** [review]
Seems fine! @cexbrayat ?

**Comment by lmiller1990** [comment]
@phobetron can you fix the linting errors?

**Comment by phobetron** [comment]
@lmiller1990 Sorry about that! I think my local has a different default line length. Should be fixed now.

**Review by cexbrayat — APPROVED** [review]
Awesome, thanks 👍

**Comment by phobetron** [comment]
@cexbrayat That makes sense. I've reduced the test code to the minimum necessary to cover the error case. Should pass everything now 😅

**Comment by phobetron** [comment]
@cexbrayat Sorry, decided to make another minor style adjustment.

**Comment by cexbrayat** [comment]
Looks great, thank you for taking the time to investigate and fix this, we really appreciate it!

**Pull request merged**